### ISSUE TITLE
MethodInfoAdvice not called when using via assembly attribute

### ISSUE THREAD

**Issue by dorianru** [ArxOne/MrAdvice#103]
```c#
using System;
using ArxOne.MrAdvice.Advice;
using MrAdviceTest;

[assembly: TestAdvice]

namespace MrAdviceTest
{
	public class TestAdvice : Attribute, IMethodAdvice, IMethodInfoAdvice
	{
		public void Advise(MethodAdviceContext context)
		{
			Console.WriteLine("begin " + context.TargetMethod.Name);
			context.Proceed();
			Console.WriteLine("end " + context.TargetMethod.Name);
		}

		public void Advise(MethodInfoAdviceContext context)
		{
			Console.WriteLine("init " + context.TargetMethod.Name);
		}
	}

	public class Program
	{
		//[TestAdvice]
		static void Test()
		{
			Console.WriteLine("test");
		}
                //[TestAdvice]
		static void Main(string[] args)
		{
			Test();
			Console.ReadKey();
		}
	}
}

```

I get the output:
```
begin Main
begin Test
test
end Test
```

But I would expect the following output (which is obtained by uncommenting the `[TestAttribute]` lines):

```
init Test
init Main
begin Main
begin Test
test
end Test
```

I am using version 2.4.8.

**Comment by picrap** [comment]
Hi,

I see two different things here:
1. Since you apply the advice at assembly, applying it to method-level won't change anything.
2. Regarding the info advice, are you sure it is not called? The fact that it does not write on console may reflect the fact that console is not initialized at the time (I honestly didn't test, it's just a guess :wink: )

Pascal.

**Comment by picrap** [comment]
If your problem is solved, please close the issue, otherwise, tell me more about it :smile:

**Comment by dorianru** [comment]
Hi, I just verified, the `Advice(MethodInfoAdviceContext)` method is indeed not called if I don't add the attribute directly to the method.

Also, the outputs I gave are the actual outputs with the attributes commented out and not commented out, respectively. You say it wouldn't change anything applying the attribute to the method level, but it does, that's why I filed this issure ;). The assembly level attribute does not seem to work at all for the InfoAdvice.
